File: src/main/java/com/lcaohoanq/nocket/domain/token/ITokenService.java
package com.lcaohoanq.nocket.domain.token;

import com.lcaohoanq.nocket.base.exception.DataNotFoundException;
import com.lcaohoanq.nocket.domain.user.User;
import java.util.UUID;
import org.springframework.stereotype.Service;

@Service

public interface ITokenService {
    Token addToken(UUID userId, String token, boolean isMobileDevice);
    Token refreshToken(String refreshToken, User user) throws Exception;
    void deleteToken(String token, User user) throws DataNotFoundException;
    Token findUserByToken(String token) throws DataNotFoundException;
}

File: server/src/main/kotlin/com/lcaohoanq/nocket/domain/token/ITokenService.kt
package com.lcaohoanq.nocket.domain.token

import com.lcaohoanq.nocket.base.exception.DataNotFoundException
import com.lcaohoanq.nocket.domain.user.User
import org.springframework.stereotype.Service
import java.util.*

@Service
interface ITokenService {
    fun addToken(userId: UUID, token: String, isMobileDevice: Boolean): Token

    @Throws(Exception::class)
    fun refreshToken(refreshToken: String, user: User): Token

    @Throws(DataNotFoundException::class)
    fun deleteToken(token: String, user: User)

    @Throws(DataNotFoundException::class)
    fun findUserByToken(token: String): Token
}
